extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -430,23 +432,23 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		// the top we'll expand nodes until we find the parent of the Notebook we're looking for
 		// get the children of root node and expand the nodes to the notebook level.
 		await this.getChildren(parentBook.rootNode);
-		// The path to the parent of the Notebook we're looking for (this is the node we're looking to expand)
-		const parentPath = notebookPath.substring(0, notebookPath.lastIndexOf(path.posix.sep));
+		// The path to the Notebook we're looking for (these are the nodes we're looking to expand)
+		const notebookFolders = notebookPath.split('/');
 		// Find number of directories between the Notebook path and the root of the book it's contained in
 		// so we know how many parent nodes to expand
 		let depthOfNotebookInBook: number = path.relative(notebookPath, parentBook.bookPath).split(path.sep).length;
 		// Walk the tree, expanding parent nodes as needed to load the child nodes until
 		// we find the one for our Notebook
-		while (depthOfNotebookInBook > 0) {
+		while (depthOfNotebookInBook > -1) {
 			// check if the notebook is available in already expanded levels.
 			bookItem = parentBook.bookItems.find(b => b.tooltip === notebookPath);
 			if (bookItem) {
 				return bookItem;
 			}
-			// Search for the parent item
-			// notebook can be inside the same folder as parent and can be in a different folder as well
-			// so check for both scenarios.
-			let bookItemToExpand = parentBook.bookItems.find(b => b.tooltip.indexOf(parentPath) > -1) ??
+			// Walk up to the parent folders one level at each iteration
```

Comment:
So I'm confused how this logic is supposed to work. Because the reason we have to do all this is that the node we're looking to expand may be nested down the structure such that we haven't loaded the book items for it or its parents yet.

With this new logic if we're starting at the bottom of the path (the parent of our Notebook) and then can't find that (for example if its parent hasn't been loaded either) then we just bail out and nothing happens. 

What I would expect to actually do here is : 

1. Start at the TOP folder
2. If our target Notebook is in the children for this folder then return
3. Otherwise does this folder contain children?
- If no then call getChildren on it and then...
- If the next node in the path is NOT in the list of children then error out (couldn't find the path)
- Otherwise set the current node we're looking at to that node and repeat the process starting from step 2

Reply:
So that's exactly what am doing, the comment might have been confusing for you. Am checking at the top folder -> notebookFolders.length - depthOfNotebookInBook essentially does that, it gives the top level folder.